ui/handle_input.go

```diff
@@ -123,12 +123,25 @@ func (win *Window) HandleInputs() {
 	// SYNC
 	if win.inputs.sync.Clicked(win.gtx) {
 		//log.Info("Sync clicked :", win.walletInfo.Synced, win.walletInfo.Syncing)
-		if win.walletInfo.Syncing {
+		if win.walletInfo.Synced || win.walletInfo.Syncing {
 			win.wallet.CancelSync()
-		} else if !win.walletInfo.Synced {
+			win.outputs.sync = win.theme.Button("Reconnect")
+		} else {
 			win.wallet.StartSync()
-			cancel := win.outputs.icons.cancel
-			win.outputs.sync = cancel
+			win.outputs.sync = win.theme.DangerButton("Cancel")
+			win.outputs.syncHeader = win.outputs.icons.cancel
+		}
+	}
+
+	if win.inputs.syncHeader.Clicked(win.gtx) {
+		//log.Info("Sync clicked :", win.walletInfo.Synced, win.walletInfo.Syncing)
+		if win.walletInfo.Synced || win.walletInfo.Syncing {
+			win.wallet.CancelSync()
+			win.outputs.sync = win.theme.Button("Reconnect")
+		} else {
+			win.wallet.StartSync()
+			win.outputs.sync = win.theme.DangerButton("Cancel")
+			win.outputs.syncHeader = win.outputs.icons.cancel
 		}
 	}
```

Comment:
Why not `if win.inputs.syncHeader.Clicked(win.gtx) || win.inputs.sync.Clicked(win.gtx)` ?